src/html-media-source.js

```diff
@@ -145,14 +155,43 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       }
 
       buffer = new VirtualSourceBuffer(this, [avcCodec, mp4aCodec]);
-      this.sourceBuffers.push(buffer);
-      return buffer;
+    } else {
+      // delegate to the native implementation
+      buffer = this.mediaSource_.addSourceBuffer(type);
     }
 
-    // delegate to the native implementation
-    buffer = this.mediaSource_.addSourceBuffer(type);
+    setActiveSourceBuffers = () => {
+      this.updateActiveSourceBuffers_();
+      // TODO buffer.one causing call stack exceeded exception
+      buffer.off('updateend', setActiveSourceBuffers);
+    };
+    // for combined audio/video tracks, we can only determine if a source buffer is
+    // active after a completed update (once it has/doesn't have videoTracks)
+    buffer.on('updateend', setActiveSourceBuffers);
```

Comment:
Is there harm in just checking/adjusting the active source buffers on every `updateend`? Cuts down on the state a bit.